docs/use_cases/advanced_retrieval_augmented_generation.md

```diff
@@ -0,0 +1,284 @@
+
+For this tutorial, we will be using a simple document cleaner that removes empty lines and extra whitespaces. If we would be working with messier data, could remove repeated substrings, such as html tags or even write custom scripts to standardize words and correct typos. Your cleaning routine will depend heavily on your specific use case and goals.
+
+```python
+from haystack.components.preprocessors import DocumentCleaner
+
+cleaner = DocumentCleaner(
+	remove_empty_lines=True,
+	remove_extra_whitespaces=True,
+	remove_repeated_substrings=False)
+```
+
+### Chunking
+Chunking refers to the process of breaking down large pieces of text into more manageable, logically coherent units. This step is crucial for efficient retrieval, as it enables the system to focus on the most relevant segments of text when generating responses. Effective chunking strategies can dramatically improve the relevance and cohesion of the generated content.
+
+```python
+from haystack.components.preprocessors import DocumentSplitter
+
+splitter = DocumentSplitter(split_by="sentence", split_length=3, split_overlap=2)
+```
+
+In this setup, we chose "sentence" for `split_by`, a choice driven to maintain granular control over the chunking process. The `split_length` is set to 3 sentences to ensure each chunk is sufficiently detailed yet concise, while a `split_overlap` of 2 sentences helps maintain context continuity between adjacent chunks. There exists no golden rule for these hyperparameters, I would advise that you try out different configurations based on the structure and type of your documents. 
+
+### Document embeddings
+Embeddings are a central component of the pre-processing stage, serving as the bridge between raw text data and the sophisticated algorithms that drive RAG systems, typically Large Language Models (LLMs). By converting words into numerical vectors, embeddings capture the semantic relationships between different terms, enabling models to understand context and generate relevant responses. Selecting the right embeddings and models is critical for achieving high-quality retrieval and generation, laying the groundwork for RAG's success.
+
+We will be using the `SentenceTransformersDocumentEbedder`, which is a sentence embedding model. This means that instead of only looking at individual words in isolation, the model considers the full context of sentences or even larger text snippets to generate embeddings. This approach allows for a deeper understanding of the text, capturing nuances and meanings that might be lost in word-level embeddings. 
+
+```python
+from haystack.components.embedders import SentenceTransformersDocumentEmbedder
+
+doc_embedder = SentenceTransformersDocumentEmbedder("BAAI/bge-small-en-v1.5")

```

Comment:
I think we should flag here, that this approach only works with certain text embedding models, possibly just link in this doc: https://docs.haystack.deepset.ai/docs/sentencetransformersdocumentembedder